src/software/new_geom/convex_polygon_test.cpp

```diff
@@ -49,62 +49,66 @@ TEST(ConvexPolygonConstructorTest, test_construct_from_initializer_list)
+                                {-2.0f, -2.0f},
+                                {-2.0f, -1.0f},
+                                {0.0f, -1.0f}}),
+                 std::invalid_argument);
+}
 
-// TODO: Re-enable this during work on Issue #1140
-// TEST(ConvexPolygonConstructorTest, test_self_intersecting_loop)
-//{
-//    /*
-//     *  3    *------------------------------------------------*
-//     *       |                                                |
-//     *       |                                                |
-//     *  2    |       *--------------------------------*       |
-//     *       |             ------           ------            |
-//     *       |                  ----      ----                |
-//     *  1    |                       ------                   |
-//     *       |                ------      ------              |
-//     *       |         ------                   -------       |
-//     *  0    *------                                   -------*
-//     *      -3      -2       -1       0       1       2       3
-//     */
-//    // Self intersecting polygon, each asterisk on the diagram is a point making up the
-//    // polygon
-//    EXPECT_THROW(ConvexPolygon({{-3.0f, 0.0f},
-//                                {-3.0f, 3.0f},
-//                                {3.0f, 3.0f},
-//                                {3.0f, 0.0f},
-//                                {-2.0f, 2.0f},
-//                                {2.0f, 2.0f}}),
-//                 std::invalid_argument);
-//}
+TEST(ConvexPolygonConstructorTest, test_self_intersecting_loop)
+{
+    /*
+     *  3    *------------------------------------------------*
+     *       |                                                |
+     *       |                                                |
+     *  2    |       *--------------------------------*       |
+     *       |             ------           ------            |
+     *       |                  ----      ----                |
+     *  1    |                       ------                   |
+     *       |                ------      ------              |
+     *       |         ------                   -------       |
+     *  0    *------                                   -------*
+     *      -3      -2       -1       0       1       2       3
+     */
+    // Self intersecting polygon, each asterisk on the diagram is a point making up the
+    // polygon
+    EXPECT_THROW(ConvexPolygon({{-3.0f, 0.0f},
+                                {-3.0f, 3.0f},
+                                {3.0f, 3.0f},
+                                {3.0f, 0.0f},
+                                {-2.0f, 2.0f},
+                                {2.0f, 2.0f}}),
+                 std::invalid_argument);
+}
+
+TEST(ConvexPolygonConstructorTest, test_ribbon_not_convex)
+{
+    EXPECT_THROW(ConvexPolygon({{0, 0}, {0, 5}, {5, 5}, {-5, 0.0f}}),
+                 std::invalid_argument);
+}
```

Comment:
Can we add tests for the convexity of degenerate polygons like mentioned in https://math.stackexchange.com/questions/1743995/determine-whether-a-polygon-is-convex-based-on-its-vertices?

So specifically a polygon where all points are the same, and a polygon where all points are on the same line.

Reply:
Done in latest